### PR TITLE
geoip: pass Var<Logger> to query functions

### LINKED ISSUE #739: geoip: pass Var<Logger> to query functions
See (and depends on) #690

### PR DESCRIPTION
**Pull request by AntonioLangiu**
As in issue #739.

(This PR is based on top of #730.)

Closes #739.

### CODE REVIEW

**Comment by AntonioLangiu** [comment]
What other events should I log? @bassosimone

File: include/measurement_kit/ooni/error.hpp [hunk 1]
```diff
@@ -18,6 +18,7 @@ MK_DEFINE_ERR(MK_ERR_OONI(5), InvalidMandatoryValueError, "")
 MK_DEFINE_ERR(MK_ERR_OONI(6), MissingRequiredHostError, "")
 MK_DEFINE_ERR(MK_ERR_OONI(7), MissingRequiredUrlError, "")
 MK_DEFINE_ERR(MK_ERR_OONI(8), CannotOpenGeoIpCountryDatabase, "")
+MK_DEFINE_ERR(MK_ERR_OONI(8), CannotOpenGeoIpCityDatabase, "")
 MK_DEFINE_ERR(MK_ERR_OONI(8), CannotOpenGeoIpAsnDatabase, "")
```
Comment (bassosimone) [inline include/measurement_kit/ooni/error.hpp]: @AntonioLangiu beware that above here you're defining three different times the same error code number.

File: src/ooni/utils.cpp [hunk 2]
```diff
@@ -76,10 +83,35 @@ ErrorOr<std::string> IPLocation::resolve_country_name(std::string ip) {
...
+    GeoIPLookup gl;
+    GeoIPRecord *gir;
+    memset (&gl, 0, sizeof(gl));
+
+    gir = GeoIP_record_by_name(gi_city, ip.c_str());
+    if (gir == nullptr) {
+        GeoIP_delete(gi_city);
+        return GenericError();
+    }
+	std::string result = gir->city;
```
Comment (bassosimone) [inline src/ooni/utils.cpp]: @AntonioLangiu beware that the above line is indented using tabs rather than spaces

File: src/ooni/utils.cpp [hunk 3]
```diff
@@ -76,10 +83,35 @@ ErrorOr<std::string> IPLocation::resolve_country_name(std::string ip) {
...
+    GeoIPRecord_delete(gir);
+    return result;
+}
+
+ErrorOr<std::string> IPLocation::resolve_asn(std::string ip,
+                                        Var<Logger> logger) {
     if (gi_asn == nullptr) {
         gi_asn = GeoIP_open(path_asn.c_str(), GEOIP_MEMORY_CACHE);
         if (gi_asn == nullptr) {
+            logger->debug("IPLocation: cannot open geoip asn database");
```
Comment (bassosimone) [inline src/ooni/utils.cpp]: I think this is a warning, not a debug message

File: src/ooni/utils.cpp [hunk 4]
```diff
@@ -108,6 +140,13 @@ ErrorOr<json> geoip(std::string ip, std::string path_country,
     if (!country_name) {
         return country_name.as_error();
     }
+	ErrorOr<std::string> city_name;
+	if (path_city != "") {
+		city_name = ip_location.resolve_city_name(ip);
+		if (!city_name) {
+			return city_name.as_error();
+		}
+	}
```
Comment (bassosimone) [inline src/ooni/utils.cpp]: @AntonioLangiu beware that the indentation of this newly added block is weird

File: src/ooni/utils.cpp [hunk 5]
```diff
@@ -116,6 +155,9 @@ ErrorOr<json> geoip(std::string ip, std::string path_country,
     node["country_code"] = country_code.as_value();
     node["country_name"] = country_name.as_value();
     node["asn"] = asn.as_value();
+	if (path_city != "") {
+		node["city_name"] = city_name.as_value();
+	}
```
Comment (bassosimone) [inline src/ooni/utils.cpp]: Same here (weird indentation)

**Comment by bassosimone** [comment]
Putting on hold because it's based on #730

**Comment by bassosimone** [comment]
> What other events should I log? @bassosimone

Perhaps you could log at debug or debug2 level the names of the files

**Comment by bassosimone** [comment]
I've added enough code to complete this (easy) pull request and I am going to merge it as soon as the automatic build process (travis-ci and circle-ci) completes.

**Pull request merged**